scripts/poweron/wlan.py

```diff
@@ -4,33 +4,19 @@
 # and send it a wake-on-LAN packet. Used for the power-on functionality as well.
 
 import socket
-import struct
-import subprocess
 import syslog
 import time
 
 import inventory
+import xcp.cmd as cmd
 import XenAPIPlugin
 
 
-def doexec(args, inputtext=None):
-    """Execute a subprocess, then return its return code, stdout and stderr"""
-    proc = subprocess.Popen(
-        args,
-        stdin=subprocess.PIPE,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        universal_newlines=True,
-        close_fds=True,
-    )
-    (stdout, stderr) = proc.communicate(inputtext)
-    rc = proc.returncode
-    return (rc, stdout, stderr)
-
-
 def find_interface_broadcast_ip(interface):
     """Return the broadcast IP address of the supplied local interface"""
-    (rc, stdout, stderr) = doexec(["ip", "address", "show", "dev", interface])
+    (rc, stdout, stderr) = cmd.runCmd(
```

Comment:
Why is this dependency preferred?

Reply:
> Why is this dependency preferred?

Introducing `xcp.cmd` is because others plugins (eg: disk-space) are also using it, and its upgrade has already been completed (refer to https://github.com/xenserver/python-libs), making it convenient for us to upgrade python3.

BTW, @liulinC had a PR for requiring xcp-python-libs in xapi.spec.